model/task/task.go

```diff
@@ -2050,31 +2055,21 @@ func (t *Task) GetDisplayStatus() string {
 // for example, if there are both successful and failed tasks, one would expect to see "failed"
 func (t *Task) displayTaskPriority() int {
```

Comment:
I don't think these changes are right; the purpose of the order of case here is that if there are multiple execution tasks, it determines which status we should see (for example if any execution tasks are in progress, then the overall display task status would be in progress). 

Additionally each status has its own number to ensure that a list of statuses would always result in the same display task status. Display task priority is used to sort the execution tasks and then we use the "highest priority" execution task status as the display task status, so by giving multiple statuses the same priority it means that display task statuses can be inconsistent.

Reply:
Ok - I had initially done this since these shared statuses hold the same rank in `addStatusColorSort` but I see what you mean.  I've separated them out now so that the sorting can be consistent, and I've added a comment that makes it clear that there is a notion of display statuses that have the same badge color.  One way to organize this is to differentiate these priorities by the singles digits, not the tens digit.

Reply:
Ah - my understanding was that we follow the thought process of "If at least one exec task failed, there's no way the display task can succeed so we mark it as preemptively failed".

Reply:
Hmm, ok, thanks for the clarity. I'd say that given this some of the work here is still relevant because of [this comment](https://github.com/evergreen-ci/evergreen/blob/main/model/task/db.go#L176) saying `displayStatusExpression ` should reflect `GetDisplayStatus` which isn't currently exactly the case. I'll revert the `displayTaskPriority` and `addStatusColorSort` if they are fine the way they are.